CMakeLists.txt

```diff
@@ -524,7 +524,7 @@ PRIVATE
 #        "CMAKE_CXX_COMPILER_ID = ${CMAKE_CXX_COMPILER_ID} "
 #        "CMAKE_CXX_COMPILER_VERSION = ${CMAKE_CXX_COMPILER_VERSION}"
 #)
-if(MSVC)
+if(${CMAKE_CXX_COMPILER_ID} MATCHES "MSVC")
```

Comment:
There is another `if(MSVC)` at line #158 that needs fixing. That is why the Appveyor CI build is failing.

Reply:
Huh, not sure why Appveyor is failing yet but changing that at 158 causes other errors that make ClangCL builds fail with: 
`clang-cl : error : argument unused during compilation: '-O0'`

Reply:
Oh I see what's happening on appveyor. Yeah, strange. Let me see if I can figure out a better solution here

Reply:
REALLY not sure why Appveyor's environment results in CMake not understanding the MSVC check